relayer/relays/beefy/beefy-relaychain-listener.go

```diff
@@ -106,11 +106,18 @@ func (li *BeefyRelaychainListener) subBeefyJustifications(ctx context.Context) e
 			}
 			log.WithField("blockHash", blockHash.Hex()).Info("Got next blockhash")
 
-			latestMMRProof, err := li.relaychainConn.GetMMRLeafForBlock(blockNumber-1, blockHash)
+			latestMMRProof, err := li.relaychainConn.GetMMRLeafForBlock(blockNumber-1, blockHash, li.config.Source.Polkadot.BeefyStartingBlock)
 			if err != nil {
 				log.WithError(err).Error("Failed get MMR Leaf")
 				return err
 			}
+
+			mmrLeafCount, err := li.relaychainConn.FetchMMRLeafCount(blockHash)
+			if err != nil {
+				log.WithError(err).Error("Failed get MMR Leaf Count")
```

Comment:
See my other comment about the error handling in `FetchMMRLeafCount`. Strictly speaking the function should not return an error if the retrieved leaf count is zero, The validation of the leaf count should be hoisted up here.

Reply:
fixed 06a2d8d